components/application-registry/cmd/applicationregistry/handlers.go

```diff
@@ -96,9 +96,9 @@ func NewSpecificationService(dynamicClient dynamic.Interface, opt *options) spec
 
 	docsTopicRepository := assetstore.NewDocsTopicRepository(resourceInterface)
 	uploadClient := upload.NewClient(opt.uploadServiceURL)
-	assetStoreService := assetstore.NewService(docsTopicRepository, uploadClient, opt.insecureAssetDownload)
+	assetStoreService := assetstore.NewService(docsTopicRepository, uploadClient, opt.insecureAssetDownload, opt.specRequestTimeout)
```

Comment:
I would suggest to use different parameters for specifying timeouts for assetstore.service and specification.service. The network connection parameters will be different for Asset Store and user provided spec url so that we need to have separate configuration.

Reply:
Done!